pandas-stubs/core/window/rolling.pyi

```diff
@@ -1,119 +1,225 @@
     Series,
 )
-from pandas.core.base import (
-    PandasObject,
-    SelectionMixin,
-)
+from pandas.core.base import SelectionMixin
+from pandas.core.groupby.ops import BaseGrouper
+from pandas.core.indexes.api import Index
 
 from pandas._typing import (
-    AggFuncType,
+    AggFuncTypeBase,
+    AggFuncTypeFrame,
+    AggFuncTypeSeriesToFrame,
     Axis,
-    Scalar,
+    NDFrameT,
+    QuantileInterpolation,
+    WindowingEngine,
+    WindowingEngineKwargs,
+    WindowingRankType,
 )
 
-class _Window(PandasObject, SelectionMixin):
-    exclusions: set[str] = ...
-    obj = ...
-    on = ...
-    closed = ...
-    window = ...
-    min_periods: int = ...
-    center = ...
-    win_type: str = ...
-    win_freq = ...
-    axis = ...
+class BaseWindow(SelectionMixin[NDFrameT], Generic[NDFrameT]):
+    exclusions: frozenset[Hashable]
+    obj: Any = ...  # Incomplete
+    on: Any = ...  # Incomplete
+    closed: Any = ...  # Incomplete
+    window: Any = ...  # Incomplete
+    min_periods: Any = ...  # Incomplete
+    center: Any = ...  # Incomplete
+    axis: Any = ...  # Incomplete
+    method: Any = ...  # Incomplete
     def __init__(
         self,
-        obj,
-        window=...,
+        obj: NDFrameT,
+        window: Any | None = ...,
         min_periods: int | None = ...,
-        center: bool | None = ...,
+        center: bool = ...,
         win_type: str | None = ...,
         axis: Axis = ...,
         on: str | Index | None = ...,
         closed: str | None = ...,
-        **kwargs,
+        method: str = ...,
+        *,
+        selection: Any | None = ...,
     ) -> None: ...
     @property
-    def is_datetimelike(self) -> bool | None: ...
+    def win_type(self): ...
     @property
-    def is_freq_type(self) -> bool: ...
+    def is_datetimelike(self) -> bool: ...
     def validate(self) -> None: ...
     def __getattr__(self, attr: str): ...
     def __iter__(self): ...
+    @overload
     def aggregate(
-        self, func: AggFuncType = ..., *args, **kwargs
-    ) -> Scalar | DataFrame | Series: ...
-    def agg(
-        self, func: AggFuncType = ..., *args, **kwargs
-    ) -> Scalar | DataFrame | Series: ...
+        self: BaseWindow[Series], func: AggFuncTypeBase, *args: Any, **kwargs: Any
+    ) -> Series: ...
+    @overload
+    def aggregate(
+        self: BaseWindow[Series],
+        func: AggFuncTypeSeriesToFrame,
+        *args: Any,
+        **kwargs: Any,
+    ) -> DataFrame: ...
+    @overload
+    def aggregate(
+        self: BaseWindow[DataFrame],
+        func: AggFuncTypeFrame,
+        *args: Any,
+        **kwargs: Any,
+    ) -> DataFrame: ...
+    agg = aggregate
 
-class Window(_Window):
-    def validate(self) -> None: ...
-    def sum(self, *args, **kwargs): ...
-    def mean(self, *args, **kwargs): ...
-    def var(self, ddof: int = ..., *args, **kwargs): ...
-    def std(self, ddof: int = ..., *args, **kwargs): ...
+class BaseWindowGroupby(BaseWindow[NDFrameT]):
+    def __init__(
```

Comment:
can't be constructed, so delete `__Init__`

Reply:
removed.